teerex/src/lib.rs

```diff
@@ -162,70 +169,143 @@ pub mod pallet {
 		// the integritee-service wants to register his enclave
 		#[pallet::call_index(0)]
 		#[pallet::weight((<T as Config>::WeightInfo::register_ias_enclave(), DispatchClass::Normal, Pays::Yes))]
-		pub fn register_ias_enclave(
+		pub fn register_sgx_enclave(
 			origin: OriginFor<T>,
-			ra_report: Vec<u8>,
-			worker_url: Vec<u8>,
+			proof: Vec<u8>,
+			worker_url: Option<Vec<u8>>,
+			attestation_method: SgxAttestationMethod,
 		) -> DispatchResultWithPostInfo {
-			log::info!("teerex: called into runtime call register_ias_enclave()");
+			log::info!("teerex: called into runtime call register_sgx_enclave()");
 			let sender = ensure_signed(origin)?;
-			ensure!(ra_report.len() <= MAX_RA_REPORT_LEN, <Error<T>>::RaReportTooLong);
-			ensure!(worker_url.len() <= MAX_URL_LEN, <Error<T>>::EnclaveUrlTooLong);
+			ensure!(proof.len() <= SGX_RA_PROOF_MAX_LEN, <Error<T>>::RaProofTooLong);
+			ensure!(
+				worker_url.clone().unwrap_or_default().len() <= MAX_URL_LEN,
+				<Error<T>>::EnclaveUrlTooLong
+			);
 			log::info!("teerex: parameter length ok");
 
-			#[cfg(not(feature = "skip-ias-check"))]
-			let enclave = Self::verify_report(&sender, ra_report)?.with_url(worker_url.clone());
+			let enclave = match attestation_method {
+				SgxAttestationMethod::Ias => {
```

Comment:
This whole function will be much more readable if we have something like:

```rust
let enclave = match attestation_method {
   SgxAttestationMethod::Ias => verify_ias_attestation(...)?,
   SgxAttestationMethod::Dcap { proxied } => verify_dcap_quote(...)?,
   SgxAttestationMethod::Skipped { proxied } => handle_skipped_attestation(...)?
}


```

Reply:
I think readability is debatable. I don't like 3 line dispatchables that do nothing else than calling a fn within the same file....and then dispatch an event. if it's only a few lines I prefer reading them in-place.
Testability would be a stronger argument IMO, but as all unit tests go to the dispatchable level directly, I saw no reason to modularize

Reply:
Hmm, I disagree here to be honest, when going down, it needs a lot of focus to even remember, whether you are still in a match branch or not, but I won't argue.